### PR TITLE
Fix camera moving in non play mode

### PR DESCRIPTION
**Pull request by bruh-9000**

### CODE REVIEW

**Comment by bruh-9000** [comment]
https://github.com/moddio/moddio2/issues/612

File: src/client.js [hunk 1]
```diff
@@ -875,10 +875,12 @@ const Client = TaroEventingClass.extend({
 	},
 
 	positionCamera: function (x, y) {
-		if (x != undefined && y != undefined) {
+		if (taro.developerMode.activeTab === "play") {
```
Comment (sloont) [inline src/client.js:878]: `taro.developerMode.activeTab` is going to be undefined if the player does not have access to editing game.

**Review by sloont — CHANGES_REQUESTED** [review]

File: src/client.js (hunk 1 above)
Comment (bruh-9000) [reply src/client.js:878]: So I can just add `|| ... == undefined` ?

**Review by Grenagar — COMMENTED** [review]
image
I suggest to use this for checking (as we use it in another places) also instead of putting check inside client.js maybe will be better idea to put in GameScene.ts

**Comment by bruh-9000** [comment]
@Grenagar I noticed though that changing zoom amount was also checked in client.js for being in the play tab, so that's why I assumed it should be in client.js

**Comment by Grenagar** [comment]
> @Grenagar I noticed though that changing zoom amount was also checked in client.js for being in the play tab, so that's why I assumed it should be in client.js

and 'instant-camera-move' is inside GameScene.ts lets think together where we should place all of them

**Comment by bruh-9000** [comment]
I mean we can go with your idea of putting it in gamescene.ts and have them in different places, maybe could leave a comment in client.js about it

**Comment by bruh-9000** [comment]
Closing and making new PR

**Comment by bruh-9000** [comment]
@Grenagar I'm looking in gamescene.ts and it already is checking for if the player is in play mode?

	taro.client.on('instant-move-camera', (x: number, y: number) => {
			if (!taro.developerMode.active || taro.developerMode.activeTab === 'play') {
			    camera.centerOn(x, y);
			}
		});

**Comment by Grenagar** [comment]
> @Grenagar I'm looking in gamescene.ts and it already is checking for if the player is in play mode?
> 
> ```
> taro.client.on('instant-move-camera', (x: number, y: number) => {
> 		if (!taro.developerMode.active || taro.developerMode.activeTab === 'play') {
> 		    camera.centerOn(x, y);
> 		}
> 	});
> ```

it check now for 'instant-move-camera', but not for normal 'position-camera'

**Comment by bruh-9000** [comment]
> > @Grenagar I'm looking in gamescene.ts and it already is checking for if the player is in play mode?
> > 
> > ```
> > taro.client.on('instant-move-camera', (x: number, y: number) => {
> > 		if (!taro.developerMode.active || taro.developerMode.activeTab === 'play') {
> > 		    camera.centerOn(x, y);
> > 		}
> > 	});
> > ```
> 
> it check now for 'instant-move-camera', but not for normal 'position-camera'

So if I copy the check for instant camera to position camera that should fix it?

**Comment by Grenagar** [comment]
> > > @Grenagar I'm looking in gamescene.ts and it already is checking for if the player is in play mode?
> > > ```
> > > taro.client.on('instant-move-camera', (x: number, y: number) => {
> > > 		if (!taro.developerMode.active || taro.developerMode.activeTab === 'play') {
> > > 		    camera.centerOn(x, y);
> > > 		}
> > > 	});
> > > ```
> > 
> > 
> > it check now for 'instant-move-camera', but not for normal 'position-camera'
> 
> So if I copy the check for instant camera to position camera that should fix it?

yes, dont forget to run 'npm run tsc' to compile js file

**Comment by bruh-9000** [comment]
https://github.com/moddio/moddio2/pull/765

**Pull request closed without merging**